### ISSUE TITLE
Build error with llvm-hs-pure-4.1.0.0

### ISSUE THREAD

**Issue by RyanGlScott** [AccelerateHS/accelerate-llvm#21]
`accelerate-llvm-1.0.0.0` won't build with `llvm-hs-pure-4.1.0.0`:

```
$ cabal install accelerate-llvm
Resolving dependencies...
Configuring accelerate-llvm-1.0.0.0...
Building accelerate-llvm-1.0.0.0...
Failed to install accelerate-llvm-1.0.0.0
Build log ( /Users/rscott/.cabal/logs/accelerate-llvm-1.0.0.0.log ):
cabal: Entering directory '/var/folders/vm/djp43kt913l500xy0z8d9s0r0000gp/T/cabal-tmp-94338/accelerate-llvm-1.0.0.0'
Configuring accelerate-llvm-1.0.0.0...
Building accelerate-llvm-1.0.0.0...
Preprocessing library accelerate-llvm-1.0.0.0...
[ 1 of 52] Compiling LLVM.AST.Type.Name ( LLVM/AST/Type/Name.hs, dist/build/LLVM/AST/Type/Name.o )
[ 2 of 52] Compiling LLVM.AST.Type.Instruction.Volatile ( LLVM/AST/Type/Instruction/Volatile.hs, dist/build/LLVM/AST/Type/Instruction/Volatile.o )
[ 3 of 52] Compiling LLVM.AST.Type.Instruction.RMW ( LLVM/AST/Type/Instruction/RMW.hs, dist/build/LLVM/AST/Type/Instruction/RMW.o )
[ 4 of 52] Compiling LLVM.AST.Type.Instruction.Compare ( LLVM/AST/Type/Instruction/Compare.hs, dist/build/LLVM/AST/Type/Instruction/Compare.o )
[ 5 of 52] Compiling LLVM.AST.Type.Instruction.Atomic ( LLVM/AST/Type/Instruction/Atomic.hs, dist/build/LLVM/AST/Type/Instruction/Atomic.o )
[ 6 of 52] Compiling LLVM.AST.Type.Flags ( LLVM/AST/Type/Flags.hs, dist/build/LLVM/AST/Type/Flags.o )
[ 7 of 52] Compiling LLVM.AST.Type.AddrSpace ( LLVM/AST/Type/AddrSpace.hs, dist/build/LLVM/AST/Type/AddrSpace.o )
[ 8 of 52] Compiling LLVM.AST.Type.Representation ( LLVM/AST/Type/Representation.hs, dist/build/LLVM/AST/Type/Representation.o )
[ 9 of 52] Compiling LLVM.AST.Type.Constant ( LLVM/AST/Type/Constant.hs, dist/build/LLVM/AST/Type/Constant.o )
[10 of 52] Compiling LLVM.AST.Type.Operand ( LLVM/AST/Type/Operand.hs, dist/build/LLVM/AST/Type/Operand.o )
[11 of 52] Compiling LLVM.AST.Type.Global ( LLVM/AST/Type/Global.hs, dist/build/LLVM/AST/Type/Global.o )
[12 of 52] Compiling LLVM.AST.Type.Instruction ( LLVM/AST/Type/Instruction.hs, dist/build/LLVM/AST/Type/Instruction.o )
[13 of 52] Compiling LLVM.AST.Type.Metadata ( LLVM/AST/Type/Metadata.hs, dist/build/LLVM/AST/Type/Metadata.o )
[14 of 52] Compiling LLVM.AST.Type.Terminator ( LLVM/AST/Type/Terminator.hs, dist/build/LLVM/AST/Type/Terminator.o )
[15 of 52] Compiling Data.Range.Range ( Data/Range/Range.hs, dist/build/Data/Range/Range.o )
[16 of 52] Compiling Data.Array.Accelerate.LLVM.Util ( Data/Array/Accelerate/LLVM/Util.hs, dist/build/Data/Array/Accelerate/LLVM/Util.o )
[17 of 52] Compiling Data.Array.Accelerate.LLVM.Target ( Data/Array/Accelerate/LLVM/Target.hs, dist/build/Data/Array/Accelerate/LLVM/Target.o )
[18 of 52] Compiling Data.Array.Accelerate.LLVM.State ( Data/Array/Accelerate/LLVM/State.hs, dist/build/Data/Array/Accelerate/LLVM/State.o )
[19 of 52] Compiling Data.Array.Accelerate.LLVM.Execute.Async ( Data/Array/Accelerate/LLVM/Execute/Async.hs, dist/build/Data/Array/Accelerate/LLVM/Execute/Async.o )
[20 of 52] Compiling Data.Array.Accelerate.LLVM.Execute.Environment ( Data/Array/Accelerate/LLVM/Execute/Environment.hs, dist/build/Data/Array/Accelerate/LLVM/Execute/Environment.o )
[21 of 52] Compiling Data.Array.Accelerate.LLVM.Execute.Marshal ( Data/Array/Accelerate/LLVM/Execute/Marshal.hs, dist/build/Data/Array/Accelerate/LLVM/Execute/Marshal.o )
[22 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Type ( Data/Array/Accelerate/LLVM/CodeGen/Type.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Type.o )
[23 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Ptr ( Data/Array/Accelerate/LLVM/CodeGen/Ptr.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Ptr.o )
[24 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Monad[boot] ( Data/Array/Accelerate/LLVM/CodeGen/Monad.hs-boot, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Monad.o-boot )
[25 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Module ( Data/Array/Accelerate/LLVM/CodeGen/Module.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Module.o )
[26 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Intrinsic ( Data/Array/Accelerate/LLVM/CodeGen/Intrinsic.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Intrinsic.o )
[27 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.IR ( Data/Array/Accelerate/LLVM/CodeGen/IR.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/IR.o )
[28 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Sugar ( Data/Array/Accelerate/LLVM/CodeGen/Sugar.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Sugar.o )
[29 of 52] Compiling Data.Array.Accelerate.LLVM.Foreign ( Data/Array/Accelerate/LLVM/Foreign.hs, dist/build/Data/Array/Accelerate/LLVM/Foreign.o )
[30 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Environment ( Data/Array/Accelerate/LLVM/CodeGen/Environment.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Environment.o )
[31 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Constant ( Data/Array/Accelerate/LLVM/CodeGen/Constant.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Constant.o )
[32 of 52] Compiling Data.Array.Accelerate.LLVM.CodeGen.Downcast ( Data/Array/Accelerate/LLVM/CodeGen/Downcast.hs, dist/build/Data/Array/Accelerate/LLVM/CodeGen/Downcast.o )

Data/Array/Accelerate/LLVM/CodeGen/Downcast.hs:444:53: error:
    Not in scope: data constructor ‘L.IEEE’
    Neither ‘LLVM.AST.Attribute’, ‘LLVM.AST.CallingConvention’, ‘LLVM.AST.Float’, ‘LLVM.AST.Global’, ‘LLVM.AST.Instruction’, ‘LLVM.AST.Name’, ‘LLVM.AST.Operand’ nor ‘LLVM.AST.Type’ exports ‘IEEE’.

Data/Array/Accelerate/LLVM/CodeGen/Downcast.hs:445:53: error:
    Not in scope: data constructor ‘L.IEEE’
    Neither ‘LLVM.AST.Attribute’, ‘LLVM.AST.CallingConvention’, ‘LLVM.AST.Float’, ‘LLVM.AST.Global’, ‘LLVM.AST.Instruction’, ‘LLVM.AST.Name’, ‘LLVM.AST.Operand’ nor ‘LLVM.AST.Type’ exports ‘IEEE’.

Data/Array/Accelerate/LLVM/CodeGen/Downcast.hs:446:53: error:
    Not in scope: data constructor ‘L.IEEE’
    Neither ‘LLVM.AST.Attribute’, ‘LLVM.AST.CallingConvention’, ‘LLVM.AST.Float’, ‘LLVM.AST.Global’, ‘LLVM.AST.Instruction’, ‘LLVM.AST.Name’, ‘LLVM.AST.Operand’ nor ‘LLVM.AST.Type’ exports ‘IEEE’.

Data/Array/Accelerate/LLVM/CodeGen/Downcast.hs:447:53: error:
    Not in scope: data constructor ‘L.IEEE’
    Neither ‘LLVM.AST.Attribute’, ‘LLVM.AST.CallingConvention’, ‘LLVM.AST.Float’, ‘LLVM.AST.Global’, ‘LLVM.AST.Instruction’, ‘LLVM.AST.Name’, ‘LLVM.AST.Operand’ nor ‘LLVM.AST.Type’ exports ‘IEEE’.
cabal: Leaving directory '/var/folders/vm/djp43kt913l500xy0z8d9s0r0000gp/T/cabal-tmp-94338/accelerate-llvm-1.0.0.0'
cabal: Error: some packages failed to install:
accelerate-llvm-1.0.0.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by tmcdonell** [comment]
I've updated the bounds on Hackage for now.

**Comment by tmcdonell** [comment]
Also, thank you for the report! Sorry I did not say this earlier ):

**Comment by tmcdonell** [comment]
Master now builds against llvm-hs-4.1

**Comment by RyanGlScott** [comment]
OK. `llvm-hs-5.0.0` was released recently, so should I keep this ticket open to track support for that, or should I just close this ticket altogether?

**Comment by tmcdonell** [comment]
Sorry. I did support llvm-hs-4.1 for a while now, but somehow never released it to hackage.

The next release is basically ready and supports 4.* and 5.0, I'm just doing final testing.

**Comment by tmcdonell** [comment]
(also I am just terrible at organising myself to release things and I am sorry)

**Comment by RyanGlScott** [comment]
[`accelerate-llvm-1.1.0.0`](http://hackage.haskell.org/package/accelerate-llvm-1.1.0.0) has been released, which supports `llvm-hs` 4.1 and 5.0.

**Comment by tmcdonell** [comment]
Sorry this took so long...
